### ISSUE TITLE
Request for a feature

### ISSUE THREAD

**Issue by prabh1601** [Jatana/FastOlympicCoding#19]
Hey Janata ! 
I just wanted to give a request for a feature such that we can also see the expected outcome / output for the cases within the plugin
. Just a little window beneath the testcases would do.

**Comment by npanuhin** [comment]
Hello!
Nice idea. In my opinion, this should be implemented as another button (next to "run" and "edit"):

```
[ test 0 ]  [ edit ]  [ run ]  [ expected ]  [ time: 69ms ]
```

and `expected` toggles between this:

```
[ test 0 ]  [ edit ]  [ run ]  [ expected ]  [ time: 69ms ]
test input

test output
[ decline ]
```

and this:

```
[ test 0 ]  [ edit ]  [ run ]  [ expected ]  [ time: 69ms ]
test input

test output

[ expected output: ]
expected output
[ decline ]
```

Also:
- `expected output` as `time: Xms` or `test X` is not clickable
- `expected` button (clickable of course) and the whole structure only appears if the target result of that case is saved

**Comment by Jatana** [comment]
Hi!

I think it is an interesting idea to add the expected output feature.
However, the test bar already contains `test number`, `edit`, `run`, and `time`, so adding the `expected` button there would be too much.
An alternative to a button could be a key binding, a menu option, or something else.
So, I am open to suggestions about the best way to integrate this feature.

**Comment by prabh1601** [comment]
Hi Nikita and Janata! 
Considering nikita its a good feature to have  a button for the same I think it should be like that only because sometimes output can stretch in cases of displaying arrays / strings etc and you always wont want to see a stretched window . But also I think Janata has a fully valid point here. In my opinion a button besides the accept/decline button can do the job.Also I would be very happy to have better ideas for the same

**Comment by npanuhin** [comment]
I added a button `expected output` after `decline` and I think it works well. But I can not figure out how the actions work. (Why is there a **list** of expected outputs for each test case?) Jatana, could you please toggle the expected output when this button is pressed?

and I think it works well. But I can't figure out how actions work. (Also, why is there a **list** of expected results for each test case?)

Jatana, could you, please, implement toggling the expected result when this button is clicked?

![изображение](https://user-images.githubusercontent.com/38539949/93489041-ba153880-f90f-11ea-98b8-7d6e8cf118e1.png)

![изображение](https://user-images.githubusercontent.com/38539949/93489091-c600fa80-f90f-11ea-8745-5e50b83f0b7f.png)

**Comment by prabh1601** [comment]
This looks neat. I guess once Janata give a green flag to this implementation and also fixes the issues that you are getting it would great one to have.

**Comment by prabh1601** [comment]
Hey Nikita and Janata ! 
Are there any updates for the same ?

**Comment by npanuhin** [comment]
Hello!
The problem is that you can actually store **multiple correct answers** for a single test.
Then the only option is to print the expected output only if there is only one such possible output. Currently, I have not figured out what to do if there are several of them...

**Comment by Jatana** [comment]
I think printing one of the correct answers could be a solution because in most cases there is only one correct answer to a test.

**Comment by prabh1601** [comment]
I agree with the point of Janata .. almost of the sample cases that are provided at a site are provided with a single test case answer but however pertaining to the point of Nikita I think just showing the first correct answer can be a viable option .

**Comment by npanuhin** [comment]
Ok, thank you. I will try to implement this in the nearest future.

**Comment by prabh1601** [comment]
Oh that great, I would be looking forward to hear from you soon .

**Comment by npanuhin** [comment]
Hello!
@Jatana, I still can't figure out how to insert text into an existing layout (i.e. just a page). Please help with this :)

PS: *If it will be a bit helpful, I [added](https://github.com/npanuhin/FastOlympicCoding) a boolean `expected_output_toggled`variable and a function `get_correct_answer` for the `Test` class*

**Comment by prabh1601** [comment]
Hi @npanuhin @Jatana !
I appreciate the efforts being put into my request and really thank you for it. 
Just wanted a little update that how are things going

**Comment by prabh1601** [comment]
Hi Guys ! Any further updates ?

**Comment by Jatana** [comment]
Sorry, I've been extremely busy this and previous month. I'll try to find some time in the upcoming days to implement the feature.

**Comment by prabh1601** [comment]
I appreciate your efforts man thank you

**Comment by prabh1601** [comment]
Hi @Jatana @npanuhin !
Just wanted to ask if there are any plans for this feature to be encorporated in the further updates ?

**Comment by liuxueyang** [comment]
I've implemented almost the same feature in this [fork](https://github.com/liuxueyang/FastOlympicCoding?tab=readme-ov-file#new-features). Can I create a PR?
